graphql/schemabuilder/schema.go

```diff
@@ -150,15 +154,38 @@ func (s *Schema) Object(name string, typ interface{}, options ...ObjectOption) *
 		if object.Methods == nil {
 			object.Methods = make(Methods)
 		}
-
-		federatedMethod := &method{}
+		federatedMethod := &method{
+			FederationType: federatedObjectType,
```

Comment:
and this to `RootObjectType`, then we also add a `ShadowObjectType` field? 
I think currently `IsRoot` and `IsShadow` shares this field which seems a bit confusion.

Reply:
ping this.